compair/static/modules/navbar/navbar-module.js

```diff
@@ -83,6 +84,30 @@ module.controller(
         $scope.showLogin = function() {
             $scope.$emit(AuthenticationService.LOGIN_REQUIRED_EVENT);
         };
+
+        $scope.impersonating = false;
```

Comment:
listen for IMPERSONATE_START_EVENT event to refresh the scope data as needed
```
        $scope.$on(Session.IMPERSONATE_START_EVENT, function() {
           $scope.getPermissions();
        });
```
and use `$scope.loggedInUser` for determining if you should display the impersonation message, etc